modules/Components/src/server/routes/CodeTableServiceRoutes.coffee

```diff
@@ -37,31 +37,33 @@ exports.getAllCodeTableValues = (req, resp) ->
 		)
 
 exports.getCodeTableValues = (req, resp) ->
-	exports.getCodeTableValuesInternal req.params.type, req.params.kind, (result) ->
+	exports.getCodeTableValuesInternal req.params.type, req.params.kind, req.query, (result) ->
 		resp.json result
 
-exports.getCodeTableValuesInternal = (type, kind, cb) ->
+exports.getCodeTableValuesInternal = (type, kind, query, cb) ->
 	if global.specRunnerTestmode
 		fullCodeTableJSON = require '../public/javascripts/spec/CodeTableJSON.js'
 		correctCodeTable = _.findWhere(fullCodeTableJSON.codes, {type:type, kind:kind})
 		cb correctCodeTable['codes']
 	else
+		# For backwards compatibility, check if the query parameter is a function
+		if typeof query == 'function'
+			console.warn('Deprecation warning: Please provide the query parameter using the new parameter order. The old order will be removed in a future release.')
+			cb = query
+			query = null
 		config = require '../conf/compiled/conf.js'
 		baseurl = "#{config.all.client.service.persistence.fullpath}ddictvalues/all/#{type}/#{kind}/codetable"
+		qs = {}
+		if query?
+			qs = query
 		request = require 'request'
 		request(
 			method: 'GET'
 			url: baseurl
+			qs: qs
 			json: true
 		, (error, response, json) =>
 			if !error && response.statusCode == 200
-				if json.length > 0 and !json[0].displayOrder? and json[0].name?
```

Comment:
I don't know why this was added. Roo always did this same sorting before passing the response and it was messing up my "best match" sort when searching by labelTextSerch so I removed it.